app/src/main/java/com/jHerscu/clearskies/ui/prefs/PreferencesScreen.kt

```diff
@@ -1 +1,267 @@
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.movableContentOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.jHerscu.clearskies.R
+import com.jHerscu.clearskies.domain.repoInterface.HomeDisplayIntervalPref
+import com.jHerscu.clearskies.domain.repoInterface.LockedThemePref
+import com.jHerscu.clearskies.domain.repoInterface.PreferenceGroup
+import com.jHerscu.clearskies.domain.repoInterface.PreferenceOrderComparator
+import com.jHerscu.clearskies.domain.repoInterface.TempUnitPref
+import com.jHerscu.clearskies.domain.repoInterface.TranslatedPrefGroup
+import com.jHerscu.clearskies.ui.components.CarouselComponent
+import com.jHerscu.clearskies.ui.components.FloatingMenuButton
+import com.jHerscu.clearskies.ui.components.PrimaryElevatedCard
+import com.jHerscu.clearskies.ui.prefs.PreferencesViewModel.Intent
+import com.jHerscu.clearskies.ui.prefs.components.PreferenceCarouselItem
+import com.jHerscu.clearskies.ui.prefs.components.PreferenceRadioComponent
+import com.jHerscu.clearskies.ui.prefs.components.PreferenceSwitchComponent
+import com.jHerscu.clearskies.ui.theme.ON_CARD_TONAL_ELEVATION_DP
+import com.jHerscu.clearskies.ui.theme.Padding
+import com.jHerscu.clearskies.utils.padding
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun PreferencesScreen(
+    drawerState: DrawerState,
+    scope: CoroutineScope,
+    windowSizeClass: WindowSizeClass,
+    modifier: Modifier = Modifier,
+    viewModel: PreferencesViewModel = hiltViewModel()
+) {
+    val context = LocalContext.current
+
+    // Local pref which resets to default when page is left and returned to
+    var isColumn by remember { mutableStateOf(true) }
+    val viewState = viewModel.viewState.collectAsStateWithLifecycle()
+
+    LaunchedEffect(viewState.value.userPrefs.prefOrderComparator) {
+        this.launch {
+            val sorted = with(viewState.value) {
+                viewModel.sortPrefGroups(
+                    translatedPrefGroups = prefGroups.map {
+                        TranslatedPrefGroup(
+                            translatedTitle = context.getString(it.titleRes),
+                            prefGroup = it
+                        )
+                    },
+                    comparator = userPrefs.prefOrderComparator
+                )
+            }
+            viewModel.updatePrefGroupOrder(sorted)
+        }
+    }
```

Comment:
sort performed here because alphabetical sort in other locales could have different results but context should not be passed into VM. Uses VM functions to maintain separation of concerns

Reply:
re-calculated when comparator changed